### DISCUSSION TITLE
Turbo Repo Cache always getting missed

### DISCUSSION THREAD

**Discussion by biratdatta** [vercel/turborepo#8440] [Help]
### Summary

So far my pipeline seems good and the code works for some reason, it is not caching during the build or if it cache it doesn't use it during the dev.
![image](https://github.com/vercel/turbo/assets/88641794/bf80875e-703c-444e-8136-3521dfbb04f6)
When I use the --dry flag to check what the command is doing, I am getting cache to be false despite it being true on the turbo.json
![image](https://github.com/vercel/turbo/assets/88641794/7f2be65a-fafe-4910-a556-6fa39b74825f)

### Additional information
```bot-app#build
  Task                           = build
  Package                        = bot-app
  Hash                           = a1ba9d5d5c350a07
  Cached (Local)                 = false
  Cached (Remote)                = false
  Directory                      = apps/bot-app
  Command                        = <NONEXISTENT>
  Outputs                        = .next/**
  Log File                       = apps/bot-app/.turbo/turbo-build.log
  Dependencies                   = @repo/configmanager#build, @repo/eslint-config#build, @repo/hooks#build, @repo/molecules#build, @repo/pages#build, @repo/provider#build, @repo/typescript-config#build, @repo/ui#build
  Dependents                     = 
  Inputs Files Considered        = 134
  .env Files Considered          = 0
  Env Vars                       = 
  Env Vars Values                = 
  Inferred Env Vars Values       = 
  Passed Through Env Vars        = 
  Passed Through Env Vars Values = 
  Resolved Task Definition       = {"outputs":["!.next/cache/**",".next/**"],"cache":true,"dependsOn":["^build"],"inputs":[],"outputMode":"full","persistent":false,"env":[],"passThroughEnv":null,"dotEnv":null,"interactive":false}
  Framework                      = nextjs```
Done in 1.65s.

**Comment by anthonyshew** [comment]
🤔 I'm not sure what might be happening here. One thing I can think of without more information:

Are all your build outputs being ignored in `.gitignore`? I've accidentally forgotten to ignore the outputs of Storybook (`.storybook` if I'm remembering correctly? Might not be...) before which results in a change for the package, resulting in cache misses due to that change. Once those get gitignored, you'd start hitting cache again.

I'm mentioning Storybook here because I see it in your configuration and it reminded me of the time I did the above, but it could be something else.

If that doesn't end up being the case, I'm not sure why that'd be happening. I'd need a reproduction to look into it further.

**Reply by biratdatta** [reply]
![image](https://github.com/vercel/turbo/assets/88641794/a3089a9f-6fb1-4f37-8109-5d866fa7294b)
Hey this is my `.gitignore` file and pretty much all my build outputs and the storybook outputs are getting `.gitignore` . If it is possible can you check out the code https://github.com/SamagraX-Stencil/stencil-web/tree/final-turborepo and tell me the further steps that I can take for the cache to work properly.  @anthonyshew

**Reply by anthonyshew** [reply]
I'm unable to get the packages installed on that repository for some reason. The `yarn install` is hanging and never finishes, so I can't run anything to debug.

**Comment by biratdatta** [comment]
Hey @anthonyshew , thanks for taking your time out. I just checked for fresh installation of our repository and it was working fine for me. Can you check that you are particullary on the final-turborepo branch `git clone -b final-turborepo https://github.com/SamagraX-Stencil/stencil-web.git` and can you install it again for one more time. 

![image](https://github.com/vercel/turbo/assets/88641794/0ed8fd1e-1620-4973-a644-3eacd5b52194)
and `yarn dev` to check two of our apps running. It would be a major help if you can help us out

**Reply by anthonyshew** [reply]
It decided to work today! 😄

I'm hitting cache when I run `turbo build` multiple times.

Do you have [the `TURBO_FORCE` environment variable set](https://turbo.build/repo/docs/reference/system-environment-variables) by chance?

I can't say I really now much of what else might be happening. This seems specific to your machine or environment somehow.

**Reply by biratdatta** [reply]
Sure will look it up with other machines in my team. Thanks for the help
